Fix CI for publishing

chris-zen opened the pull request:
This PR provides the following novelties:
- Using pyproject.toml
- Support for uv
- Supporting cython in the build
- A Makefile with common development tasks (checking code, building packages and docker, publishing packages and docker, ...)
- Producing and publishing packages for all the supported python versions (3.8, 3.9, 3.10, 3.11) and platforms (linux, macos)

chris-zen committed:
Fix CI for publishing

FedericaBrando commented on Makefile:
```diff
@@ -75,51 +100,45 @@ check-docker:
-	@if [[ -z "$(PYPI_TOKEN)" ]]; then \
-		echo "$(BOLDRED)==> Missing PYPI_TOKEN !!!$(RESET)"; \
+publish-dist: uv-installed
+	@echo "$(BOLDGREEN)Publishing OncodriveCLUSTL $(BOLDYELLOW)$(call version)$(BOLDGREEN) to PyPI ...$(RESET)"
```
```suggestion
	@echo "$(BOLDGREEN)Publishing OncodriveFML $(BOLDYELLOW)$(call version)$(BOLDGREEN) to PyPI ...$(RESET)"
```

FedericaBrando reviewed (commented):
Looks great, I just spot a typo in the make that I've commented. :) 

thanks a lot! 😄

FedericaBrando commented:
I was checking the README.md and this part is not contained in the Makefile:

https://github.com/bbglab/oncodrivefml/blob/628fece07239ab83f1c57923187e0ab59d27c12b/README.md?plain=1#L84-L90C4

while in the Makefile the `install-dev` is not present.

chris-zen commented:
> I was checking the README.md and this part is not contained in the Makefile:
> 
> https://github.com/bbglab/oncodrivefml/blob/628fece07239ab83f1c57923187e0ab59d27c12b/README.md?plain=1#L84-L90
> 
> while in the Makefile the `install-dev` is not present. How would the readme change in this case?

Fixed. Added `build-dev` to the `Makefile` and updated the README.md
